bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1580,9 +1582,14 @@ OperationResult PreProcessor::launchReqPreProcessing(const PreProcessRequestMsgS
   auto preProcessResult = static_cast<OperationResult>(request.outExecutionStatus);
   resultLen = request.outActualReplySize;
   if (preProcessResult != OperationResult::SUCCESS) {
-    LOG_ERROR(logger(),
-              "Pre-execution failed" << KVLOG(
-                  cid, clientId, reqOffsetInBatch, reqSeqNum, (uint32_t)preProcessResult, resultLen));
+    strcpy(preProcessResultBuffer, "Error in execution request");
+    resultLen = strlen(preProcessResultBuffer);
+    // Append the conflict detection block id and add its size to the resulting length.
+    memcpy(preProcessResultBuffer + resultLen, reinterpret_cast<char *>(&blockId), sizeof(uint64_t));
+    resultLen += sizeof(uint64_t);
+    LOG_DEBUG(logger(),
+              "Pre-execution operation successfully completed with errored result"
+                  << KVLOG(cid, reqSeqNum, clientId, reqOffsetInBatch, blockId, (uint32_t)preProcessResult));
     return preProcessResult;
   }
   if (request.outActualReplySize == 0) {
```

Comment:
Here we need to fill the preProcessResultBuffer with en error "Empty execution data received" and fill the block at the end of the function, too.

Reply:
done